src/main/java/Parser.java

```diff
@@ -0,0 +1,166 @@
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+public class Parser {
+    private static final String ERROR_DESCRIPTION = "OOPS!!! The description cannot be empty.";
+    private static final String ERROR_SEARCH_TERM = "OPPS!!! The search term for find cannot be empty.";
+
+    /**
+     * Returns a Command object based on the fullCommand given
+     * @param fullCommand (command entered by the user)
+     * @return Command comamnd
+     * @throws DukeException
+     * @throws DukeDeadlineException
+     */
+    public static Command parse(String fullCommand) throws DukeException , DukeDeadlineException {
+        fullCommand = fullCommand.trim();
+        int firstSpace = fullCommand.indexOf(" ");
+        String keyword = firstSpace == -1 ? fullCommand : fullCommand.substring(0, firstSpace).toLowerCase();
```

Comment:
The CS2103 Java Coding Standard gives the following examples of acceptable ternary statements:
```
alpha = (aLongBooleanExpression) ? beta : gamma;
alpha = (aLongBooleanExpression)
        ? beta
        : gamma;
```

Consider enclosing the boolean term within parentheses, as in the following:
```
String keyword = (firstSpace == -1) ? fullCommand : fullCommand.substring(0, firstSpace).toLowerCase();
```